docs/debug.md

```diff
@@ -0,0 +1,173 @@
+      az group create -n $RESOURCEGROUP --location westus
+    ```
+2. Create the Azure Key Vault
+    - location will be set to westus. This Resource will be placed inside our previously created Resource Group `AzureProviderDemoGroup`.
+    ```bash
+    # $RANDOM will generate a random between 0 and 32767
+    KEYVAULT_NAME=keyvault-demo-$(echo $RANDOM)
+
+    az keyvault create -n $KEYVAULT_NAME -g $RESOURCEGROUP --location westus
+    ```
+
+## Service Principal Generation
+
+A service principal needs to be generated to create a user with limited permissions to access the keys in the keyvault. We will set keys in the Azure Key Vault through our **original** admin privileges, and then **Read** the keys through the **generated service principal permissions.**
+
+1. Receive the Key Vault Resource ID
+    ```bash
+    KEYVAULT_RESOURCE_ID=$(az keyvault show -n $KEYVAULT_NAME --query id -o tsv)
+    ```
+
+2. Generate The Service Principal and query the Client ID
+    ```bash
+    CLIENT_ID=$(az ad sp create-for-rbac  --name $KEYVAULT_NAME --role Reader --scopes $KEYVAULT_RESOURCE_ID --query appId -o tsv)
+    ```
+    - this is your KEYVAULT_CLIENT_ID
+3. Reset the Service Principal Password(Client Secret) and store it in `CLIENT_SECRET`
+    ```bash
+    CLIENT_SECRET=$(az ad sp credential reset --name $CLIENT_ID --credential-description "APClientSecret" --query password -o tsv)
+    ```
+    - This is your  `KEYVAULT_CLIENT_SECRET`
+4. Set the Azure Key Vault Policy to `Get` (Read Only) using the Service Principal just now created
+
+        az keyvault set-policy -n $KEYVAULT_NAME --secret-permissions get --spn $CLIENT_ID
+
+5. While still under admin privileges, add the `CLIENT_SECRET` and `CLIENT_ID` to the Key Vault as secrets.
+      ```bash
+      az keyvault secret set --vault-name $KEYVAULT_NAME --name "clientSecret" --value $CLIENT_SECRET
+      az keyvault secret set --vault-name $KEYVAULT_NAME --name "clientId" --value $CLIENT_ID
+      ```
+
+## Set Up Debug Directory
+
+1. create a secrets environment file (`secrets.env`) inside your `/debug` directory.
+    ```bash
+    # debug/secrets.env
+
+    KEYVAULT_RESOURCE_GROUP=<YOUR_RESOURCE_GROUP>
+    KEYVAULT_NAME=<YOUR_KEYVAULT_NAME>
+    KEYVAULT_CLIENT_ID=<YOUR_KEYVAULT_CLIENT_ID>
+    KEYVAULT_CLIENT_SECRET=<YOUR_KEYVAULT_CLIENT_SECRET>
+    AZURE_SUBSCRIPTION_ID=<YOUR_AZURE_SUBSCRIPTION_ID>
+    AZURE_TENANT_ID=<YOUR_AZURE_TENANT_ID>
+    ```
+2. Query the values for each of the keys above.
+    ```bash
+    # KEYVAULT_RESOURCE_GROUP
+    $ echo $RESOURCEGROUP
+
+    # KEYVAULT_NAME
+    $ echo $KEYVAULT_NAME
+
+    # KEYVAULT_CLIENT_ID
+    $ echo $CLIENT_ID
+
+    # KEYVAULT_CLIENT_SECRET
+    $ echo $CLIENT_SECRET
+
+    # AZURE_SUBSCRIPTION_ID
+    $ AZURE_SUBSCRIPTION_ID=$(az account show --query id -o tsv)
+    $ echo $AZURE_SUBSCRIPTION_ID
+
+    # AZURE_TENANT_ID
+    $ AZURE_TENANT_ID=$(az account show --query tenantId -o tsv)
+    $ echo $AZURE_TENANT_ID
+    ```
```

Comment:
Swap these around and simplify. Make env names consistent - this will allow you to remove comments.